build-system/tasks/get-zindex.js

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+var fs = require('fs');
+var gulp = require('gulp-help')(require('gulp'));
+var postcss = require('postcss');
+var table = require('text-table');
+var through = require('through2');
+var util = require('gulp-util');
+
+var tableHeaders = [
+ ['selector', 'z-index', 'file'],
+ ['---', '---', '---'],
+];
+
+var tableOptions = {
+  align: ['l', 'l', 'l'],
+  hsep: '   |   ',
+};
+
+
+/**
+ * @param {!Object<string, !Array<number>} acc accumulator object for selectors
+ * @param {!Rules} css post css rules object
+ */
+function zIndexCollector(acc, css) {
+  css.walkRules(rule => {
+    rule.walkDecls(decl => {
+      // Split out multi selector rules
+      var selectorNames = rule.selector.replace('\n', '');
+      selectorNames = selectorNames.split(',');
+      if (decl.prop == 'z-index') {
+        selectorNames.forEach(selector => {
+          if (!acc[selector]) {
+            acc[selector] = [];
+          }
+          acc[selector].push(decl.value);
+        });
+      }
+    });
+  });
+}
+
+/**
+ * @param {!Object<string, !Object<string, !Array<number>} filesData
+ *    accumulation of files and the rules and z index values.
+ * @param {!Vinyl} file vinyl fs object
+ * @param {string} enc encoding value
+ * @param {function(err: ?Object, data: !Vinyl|string)} cb chunk data through
+ */
+function onFileThrough(filesData, file, enc, cb) {
+  if (file.isNull()) {
+    cb(null, file);
+    return;
+  }
+
+  if (file.isStream()) {
+    cb(new util.PluginError('size', 'Stream not supported'));
+    return;
+  }
+
+  var selectors = filesData[file.relative] = Object.create(null);
+
+  postcss([zIndexCollector.bind(null, selectors)])
+      .process(file.contents.toString(), {
+        from: file.relative
+      }).then(res => {
+        cb(null, file);
+      });
+}
+
+/**
+ * @param {!Object<string, !Object<string, !Array<number>} filesData
+ *    accumulation of files and the rules and z index values.
+ * @param {function()} cb callback to end the stream
+ */
+function onFileThroughEnd(filesData, cb) {
+  var rows = [];
+  rows.unshift.apply(rows, tableHeaders);
+  Object.keys(filesData).sort().forEach((fileName, fileIdx) => {
+    var selectors = filesData[fileName];
+    Object.keys(selectors).sort().forEach((selectorName, selectorIdx) => {
+      var zIndexes = selectors[selectorName];
+      var row = [selectorName, zIndexes.join(', '), fileName];
+      rows.push(row);
+    });
+  });
+  var tbl = table(rows, tableOptions);
```

Comment:
yeah, the reason why i didn't do that is its possible to have multiple z-indexes on a single row if for example one declared a z-index in one rule on the same file and then reassigned a new z-index somewhere down the file. (i know its stupid and we probably would catch it in review, but its just possible in css thats all). will sort